src/tiledb/cloud/soma/ingest.py

```diff
@@ -103,87 +102,54 @@ def run_ingest_workflow_udf(
 
     logger = get_logger_wrapper(level=logging_level)
 
-    h5ad_ingest = None
-    collector = None
-
+    input_files = []
     vfs = tiledb.VFS(config=extra_tiledb_config)
 
     if vfs.is_dir(input_uri):
-        if dry_run:
-            name = "dry-run-h5ad-files"
-        else:
-            name = "ingest-h5ad-files"
-
-        grf = dag.DAG(
-            name=name,
-            mode=dag.Mode.BATCH,
-            namespace=carry_along.get("namespace", namespace),
-        )
-
-        collector = grf.submit(
-            lambda output_uri: output_uri,
-            output_uri=output_uri,
-        )
-
-        for entry_input_uri in vfs.ls(input_uri):
-            logger.debug("Processing folder item: entry_input_uri=%r", entry_input_uri)
+        for input_item in vfs.ls(input_uri):
+            logger.debug(
+                "Filtering directory items: input_uri=%r, input_item=%r, pattern=%r",
+                input_uri,
+                input_item,
+                pattern,
+            )
 
             # Subdirectories/subfolders can't be ingested.
-            if vfs.is_dir(entry_input_uri):
-                logger.info(
-                    "Skipping sub-directory: entry_input_uri=%r", entry_input_uri
-                )
-                continue
-
-            base = os.path.basename(entry_input_uri)
-            base, _ = os.path.splitext(base)
-
-            entry_output_uri = output_uri + "/" + base
-            if not output_uri.endswith("/"):
-                entry_output_uri += "/"
-            entry_output_uri += base
```

Comment:
I don't understand why we appended `base` to the output path twice. Now we only do it once.

Reply:
I think it's not a requirement